### PR TITLE
ZBUG-2128:- Added enableDisableChecks to disable zimbraFeatureWebClientEnabled, zimbraPrefImapEnabled & zimbraPrefPop3Enabled is not present on LDAP.

### PR DESCRIPTION
**Pull request by madhavdhuppe**

### CODE REVIEW

**Review by BhavinBathani — APPROVED** [review]
LGTM

File: WebRoot/js/zimbraAdmin/cos/view/ZaCosXFormView.js [hunk 1]
```diff
@@ -1170,11 +1178,17 @@ ZaCosXFormView.myXFormModifier = function(xFormObject, entry) {
...
+                    label:ZaMsg.LBL_zimbraPrefImapEnabled, trueValue:"TRUE", falseValue:"FALSE",
+                    enableDisableChecks: [ function() {
+                        return ZaCosXFormView.isAttributeDefined.call(this, ZaCos.A_zimbraPrefImapEnabled);
+                    }]
                 },
                 {ref:ZaCos.A_zimbraPrefPop3Enabled, type:_CHECKBOX_,
                     msgName:ZaMsg.LBL_zimbraPrefPop3Enabled,
-                    label:ZaMsg.LBL_zimbraPrefPop3Enabled, trueValue:"TRUE", falseValue:"FALSE"
+                    label:ZaMsg.LBL_zimbraPrefPop3Enabled, trueValue:"TRUE", falseValue:"FALSE",
+                    enableDisableChecks: [ function() {
```
Comment (silentsakky) [inline WebRoot/js/zimbraAdmin/cos/view/ZaCosXFormView.js:1189]: is this disabling the option or hiding it?

File: WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js [hunk 2]
```diff
@@ -2711,10 +2719,15 @@ textFieldCssClass:"admin_xform_number_input"}
...
-                                },
+                                    colSpan:2,
+                                    enableDisableChecks : [ function() {
+                                        return ZaAccountXFormView.isAttributeDefined.call(this, ZaCos.A_zimbraPrefImapEnabled);
+                                    }]},
                                 {ref:ZaAccount.A_zimbraPrefPop3Enabled,colSpan:2,
-                                    type:_SUPER_CHECKBOX_, resetToSuperLabel:ZaMsg.NAD_ResetToCOS, msgName:ZaMsg.LBL_zimbraPrefPop3Enabled,checkBoxLabel:ZaMsg.LBL_zimbraPrefPop3Enabled, trueValue:"TRUE", falseValue:"FALSE"}
+                                    type:_SUPER_CHECKBOX_, resetToSuperLabel:ZaMsg.NAD_ResetToCOS, msgName:ZaMsg.LBL_zimbraPrefPop3Enabled,checkBoxLabel:ZaMsg.LBL_zimbraPrefPop3Enabled, trueValue:"TRUE", falseValue:"FALSE", 
+                                    enableDisableChecks : [ function() {
+                                        return ZaAccountXFormView.isAttributeDefined.call(this, ZaCos.A_zimbraPrefImapEnabled);
```
Comment (silentsakky) [inline WebRoot/js/zimbraAdmin/accounts/view/ZaAccountXFormView.js:2729]: ZaCos.A_zimbraPrefPop3Enabled

File: WebRoot/js/zimbraAdmin/cos/view/ZaCosXFormView.js (hunk 1 above)
Comment (silentsakky) [reply WebRoot/js/zimbraAdmin/cos/view/ZaCosXFormView.js:1189]: this disables option, instead we should hide it

Comment (madhavdhuppe) [reply WebRoot/js/zimbraAdmin/cos/view/ZaCosXFormView.js:1189]: sure will hide it

**Review by silentsakky — APPROVED** [review]

**Review by BhavinBathani — APPROVED** [review]

**Pull request merged**